Packs/GSuiteSecurityAlertCenter/Integrations/GSuiteSecurityAlertCenter/GSuiteSecurityAlertCenter.py

```diff
@@ -0,0 +1,719 @@
+
+    :type first_fetch: string
+    :param first_fetch: date
+
+    :return: raise ValueError if validation fails, else return parsed date.
+    :rtype: str
+    """
+    date_time = arg_to_datetime(first_fetch, is_utc=True, settings={'STRICT_PARSING': True})
+    if date_time:
+        create_time = date_time.strftime(DATE_FORMAT)
+    else:
+        raise ValueError(MESSAGES['INVALID_PARAM_VALUE_ERROR'].format('first fetch time interval'))
+
+    current_time = datetime.utcnow()
+    date_time_obj = datetime.strptime(create_time, DATE_FORMAT)
+
+    if date_time_obj > current_time:
+        raise ValueError(MESSAGES['INVALID_PARAM_VALUE_ERROR'].format('first fetch time interval'))
+
+    return create_time
+
+
+def validate_params_for_fetch_incidents(params: Dict[str, Any], last_run: Dict) -> Tuple[Dict[str, Any], str]:
+    """
+    Validates parameters for fetch-incidents command.
+
+    :param params: parameters dictionary.
+    :param last_run: A dict with a key containing the latest incident created time we got
+        from last fetch
+
+    :return: A tuple containing two elements:
+            params (``Dict[str, int]``): fetch incident parameters.
+            last_fetch (``str``): updated last_fetch.
+    """
+    params = GSuiteClient.remove_empty_entities(params)
+
+    # get user provided fetch_first interval
+    first_fetch = params.get('first_fetch')
+    first_fetch = DEFAULT_FIRST_FETCH if not first_fetch else first_fetch
+
+    # Validate the first_fetch value
+    create_time = validate_date(first_fetch)
+
+    # get the last_fetch value
+    last_fetch = last_run['last_fetch'] if last_run.get('last_fetch') else create_time
+
+    # get the user provided max_fetch value
+    max_fetch = params.get('max_fetch', DEFAULT_MAX_FETCH)
+    max_fetch = GSuiteClient.validate_get_int(max_fetch, limit=1000, message=MESSAGES['MAX_INCIDENT_ERROR'])
+
+    next_page_token = last_run.get('next_page_token', '')
+
+    # set the filter with last_fetch value
+    alert_filter = f'createTime>"{last_fetch}"'
+
+    # get user provided filter value and validate it
+    advance_filter = params.get('filter', '').replace("'", '"')
+    if 'createTime' in advance_filter or 'create_time' in advance_filter:
+        raise ValueError(MESSAGES['INVALID_FILTER'])
+    if advance_filter != '':
+        alert_filter += f' AND {advance_filter}'
+
+    if 'type' not in alert_filter:
+        alert_types = params.get('alert_type', '').split(',')
+
+        for index in range(0, len(alert_types)):
+            alert_types[index] = alert_types[index].replace("\"", "").replace("'", "").strip()
+            if index == 0 and alert_types[index] != '':
+                alert_filter += f' AND type="{alert_types[index]}"'
+            elif alert_types[index] != '':
+                alert_filter += f' OR type="{alert_types[index]}"'
+
+    # If next_page_token is present replace filter with filter present in last_run as API does not support
+    # change in filter.
```

Comment:
We have kept this logic because whenever we have `next_page_token` from an API call and we pass that param to fetch next page data with changed filter param value, the server responds with "Invalid Argument" `400` error. 

So to handle this in fetch-incident we have kept the logic that until next_page_token is present use the previous filter, if next_page_token is not present the latest filter value would be used from configuration.